### PR TITLE
Added error handling and updated error responses for application operations

### PR DESCRIPTION
**Pull request by sivakaruna**
# Description

This PR includes the following proposed change(s):

- DF-247, DF-175

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

Local tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: digitalforms-api/src/test/java/ca/bc/gov/open/pssg/rsbc/digitalforms/controller/ApplicationFormControllerTests.java [hunk 1]
```diff
@@ -42,59 +44,65 @@ public void init() {
 		MockitoAnnotations.initMocks(this);
 	}
 
+	// Need to update
 	@DisplayName("Get success - ApplicationFormController")
 	@Test
 	void getFormSuccess() {
```
Comment (shaunmillargov) [inline digitalforms-api/src/test/java/ca/bc/gov/open/pssg/rsbc/digitalforms/controller/ApplicationFormControllerTests.java:50]: Just curious why the method name is getFormSuccess but we're checking for a NOT_FOUND in the response?

Comment (sivakaruna) [reply digitalforms-api/src/test/java/ca/bc/gov/open/pssg/rsbc/digitalforms/controller/ApplicationFormControllerTests.java:50]: I need to update the unit tests, I could comment it out instead.

Comment (shaunmillargov) [reply digitalforms-api/src/test/java/ca/bc/gov/open/pssg/rsbc/digitalforms/controller/ApplicationFormControllerTests.java:50]: Do what works for you as we don't forget about it. Maybe leave a "TODO"

Comment (sivakaruna) [reply digitalforms-api/src/test/java/ca/bc/gov/open/pssg/rsbc/digitalforms/controller/ApplicationFormControllerTests.java:51]: Added a todo

**Pull request merged**